### DISCUSSION TITLE
F score and IOU calculation

### DISCUSSION THREAD

**Discussion by jehanjen** [ultralytics/ultralytics#8076] [Q&A]
hii All..
i would like to ask if its possible to print ION and F score result with the trainig last result,i mean with P,R,MAP0.5 MAP 0.9??

**Comment by pderrenger** [comment]
Hello! 👋

Yes, it's possible to calculate and print additional metrics like IoU (Intersection over Union) and F1 score along with Precision (P), Recall (R), mAP @jehanjen.5, and mAP@0.5:0.95 during training or validation. These metrics are not printed by default, but you can modify the code to calculate and display them as needed.

For IoU, it's typically calculated per class and then averaged, while the F1 score is the harmonic mean of precision and recall. See https://en.wikipedia.org/wiki/F-score for details on computing F1 from P and R. If you're comfortable with modifying the code, you can add these calculations in the part of the code that handles metric computation.

If you need further assistance with this, feel free to ask! 😊
